app/controllers/products_controller.rb

```diff
@@ -36,6 +41,21 @@ def set_product
     @product = Product.friendly.find(product_id)
   end
 
+  def can_view_product
+    # if practice is not published
+    unless @product.published
+      unauthorized_response if current_user.blank?
+      prevent_product_permissions if current_user.present?
+    end
+  end
+
+  def prevent_product_permissions
```

Comment:
whoops got this covered below with `check_product_permissions`

Reply:
Nice catch. Less is more!